src/main/java/seedu/address/model/facilitator/UniqueFacilitatorList.java

```diff
@@ -89,29 +89,45 @@ public void remove(Facilitator toRemove) {
-            if (facilitator.getModuleCodes().contains(toRemove)) {
-                if (facilitator.getModuleCodes().size() == 1) {
-                    facilitatorsToDelete.add(facilitator);
-                } else {
-                    facilitatorsToEdit.add(facilitator);
-                }
-            }
-        }
-        for (Facilitator facilitator : facilitatorsToDelete) {
-            remove(facilitator);
-        }
-        for (Facilitator facilitator : facilitatorsToEdit) {
-            Set<ModuleCode> moduleCodes = new HashSet<>(facilitator.getModuleCodes());
-            moduleCodes.remove(toRemove);
-            Facilitator editedFacilitator = new Facilitator(facilitator.getName(), facilitator.getPhone(),
-                    facilitator.getEmail(), facilitator.getOffice(), moduleCodes);
-            setFacilitator(facilitator, editedFacilitator);
-        }
+        //        List<Facilitator> facilitatorsToDelete = new ArrayList<>();
+        //        List<Facilitator> facilitatorsToEdit = new ArrayList<>();
+        //        for (Facilitator facilitator : internalList) {
+        //            if (facilitator.getModuleCodes().contains(toRemove)) {
+        //                if (facilitator.getModuleCodes().size() == 1) {
+        //                    facilitatorsToDelete.add(facilitator);
+        //                } else {
+        //                    facilitatorsToEdit.add(facilitator);
+        //                }
+        //            }
+        //        }
+        //        for (Facilitator facilitator : facilitatorsToDelete) {
+        //            remove(facilitator);
+        //        }
+        //        for (Facilitator facilitator : facilitatorsToEdit) {
+        //            Set<ModuleCode> moduleCodes = new HashSet<>(facilitator.getModuleCodes());
+        //            moduleCodes.remove(toRemove);
+        //            Facilitator editedFacilitator = new Facilitator(facilitator.getName(), facilitator.getPhone(),
+        //                    facilitator.getEmail(), facilitator.getOffice(), moduleCodes);
+        //            setFacilitator(facilitator, editedFacilitator);
+        //        }
+
+        List<Facilitator> replacementList = new ArrayList<>();
+        internalList.stream()
+                .filter(facilitator -> {
+                    Set<ModuleCode> temp = facilitator.getModuleCodes();
+                    return temp.contains(toRemove) && temp.size() > 1;
+                })
+                .map(facilitator -> {
+                    Set<ModuleCode> moduleCodes = new HashSet<>(facilitator.getModuleCodes());
+                    moduleCodes.remove(toRemove);
+                    return new Facilitator(facilitator.getName(), facilitator.getPhone(),
+                            facilitator.getEmail(), facilitator.getOffice(), moduleCodes);
+                })
+                .forEach(replacementList::add);
+
+        setFacilitators(replacementList);
```

Comment:
What about facilitators whose module code does not contain toRemove? I think have to add them to the replacementList too

Reply:
whoops forgot that part